src/main/java/io/proximax/sdk/model/transaction/builder/TransactionBuilder.java

```diff
@@ -0,0 +1,204 @@
+   
+   /**
+    * @param version the version to set
+    */
+   public B version(Integer version) {
+      this.version = version;
+      return self();
+   }
+
+   /**
+    * @param networkType the networkType to set
+    */
+   public B networkType(NetworkType networkType) {
+      this.networkType = networkType;
+      return self();
+   }
+
+   /**
+    * @param deadline the fixed deadline to use
+    */
+   public B deadline(TransactionDeadline deadline) {
+      this.deadline = Optional.of(deadline);
+      return self();
+   }
+
+   /**
+    * @param deadlineMillis the deadline specified as duration in milliseconds after {@link TransactionBuilder#build()}
+    * is invoked
+    */
+   public B deadlineDuration(BigInteger deadlineMillis) {
+      this.deadlineMillis = Optional.of(deadlineMillis);
+      return self();
+   }
+
+   /**
+    * @param maxFee the maxFee to set
+    */
+   public B maxFee(BigInteger maxFee) {
+      this.maxFee = Optional.of(maxFee);
+      return self();
+   }
+
+   /**
+    * @param signature the signature to set
+    */
+   public B signature(Optional<String> signature) {
+      this.signature = signature;
+      return self();
+   }
+
+   /**
+    * @param signer the signer to set
+    */
+   public B signer(Optional<PublicAccount> signer) {
+      this.signer = signer;
+      return self();
+   }
+
+   /**
+    * @param transactionInfo the transactionInfo to set
+    */
+   public B transactionInfo(Optional<TransactionInfo> transactionInfo) {
+      this.transactionInfo = transactionInfo;
+      return self();
+   }
+
+   /**
+    * @param feeCalculationStrategy the feeCalculationStrategy to set
+    */
+   public B feeCalculationStrategy(FeeCalculationStrategy feeCalculationStrategy) {
+      this.feeCalculationStrategy = Optional.of(feeCalculationStrategy);
+      return self();
+   }
+
+   //-------------------------------------- getters ---------------------------------------//
+   
+   /**
+    * @return the type
+    */
+   public TransactionType getType() {
+      return type;
+   }
+
+   /**
+    * @return the version
+    */
+   public Integer getVersion() {
+      return version;
+   }
+
+   /**
+    * @return the networkType
+    */
+   public NetworkType getNetworkType() {
+      return networkType;
+   }
+
+   /**
+    * @return the maxFee
+    */
+   public Optional<BigInteger> getMaxFee() {
+      return maxFee;
+   }
+
+   /**
+    * @return the signature
+    */
+   public Optional<String> getSignature() {
+      return signature;
+   }
+
+   /**
+    * @return the signer
+    */
+   public Optional<PublicAccount> getSigner() {
+      return signer;
+   }
+
+   /**
+    * @return the transactionInfo
+    */
+   public Optional<TransactionInfo> getTransactionInfo() {
+      return transactionInfo;
+   }
+
+   /**
+    * @return the feeCalculationStrategy
+    */
+   public Optional<FeeCalculationStrategy> getFeeCalculationStrategy() {
```

Comment:
I think that FeeCalculationStrategy is not part of transaction. It is part of transaction builder

Reply:
hm that is good point BUT transaction is immutable so fee would need to be calculated before transaction is created. That would make size calculation external, complicating the whole API. Or builder would need to be part of transaction which feels a bit worse than strategy in my opinion :)
I am willing to accept that transaction instance is using either explicit fee or strategy to calculate fee. That way the API is available via builder and via constructor (as builder is not mandatory)

Reply:
I decided to externalize the size calculation - there will be static method with required parameters. I will provide static constructor to create transaction by specifying fee calculation to make the api less clunky. builder will hide it implicitly. Change will be made in next commit